prefect_gcp/worker.py

```diff
@@ -0,0 +1,642 @@
+            )
+
+    @validator("job_body")
+    def _ensure_job_includes_all_required_components(cls, value: Dict[str, Any]):
+        """
+        Ensures that the job body includes all required components.
+        """
+        patch = JsonPatch.from_diff(value, _get_base_job_body())
+        missing_paths = sorted([op["path"] for op in patch if op["op"] == "add"])
+        if missing_paths:
+            raise ValueError(
+                "Job is missing required attributes at the following paths: "
+                f"{', '.join(missing_paths)}"
+            )
+        return value
+
+    @validator("job_body")
+    def _ensure_job_has_compatible_values(cls, value: Dict[str, Any]):
+        """Ensure that the job body has compatible values."""
+        patch = JsonPatch.from_diff(value, _get_base_job_body())
+        incompatible = sorted(
+            [
+                f"{op['path']} must have value {op['value']!r}"
+                for op in patch
+                if op["op"] == "replace"
+            ]
+        )
+        if incompatible:
+            raise ValueError(
+                "Job has incompatible values for the following attributes: "
+                f"{', '.join(incompatible)}"
+            )
+        return value
+
+
+class CloudRunWorkerVariables(BaseVariables):
+    """
+    Default variables for the Cloud Run worker.
+
+    The schema for this class is used to populate the `variables` section of the default
+    base job template.
+    """
+
+    region: str = Field(..., description="The region where the Cloud Run Job resides.")
+    credentials: Optional[GcpCredentials] = Field(
+        title="GCP Credentials",
+        default_factory=GcpCredentials,
+        description="The GCP Credentials used to initiate the "
+        "Cloud Run Job. If not provided credentials will be "
+        "inferred from the local environment.",
+    )
+    image: Optional[str] = Field(
+        default=None,
+        title="Image Name",
+        description=(
+            "The image to use for a new Cloud Run Job. "
+            "See https://cloud.google.com/run/docs/deploying#images"
+            "for supported image registries. "
+            "If not set, the latest Prefect image will be used."
+        ),
+        example="docker.io/prefecthq/prefect:2-latest",
+    )
+    cpu: Optional[str] = Field(
+        default=None,
+        title="CPU",
+        description=(
+            "The amount of compute allocated to the Cloud Run Job. "
+            "The int must be valid based on the rules specified at "
+            "https://cloud.google.com/run/docs/configuring/cpu#setting-jobs."
+            "1000m = 1 CPU."
+        ),
+        example="1000m",
+        regex=r"^(\d*000)m$",
+    )
+    memory: Optional[str] = Field(
+        default=None,
+        title="Memory",
+        description="The amount of memory allocated to the Cloud Run Job. See"
+        "https://cloud.google.com/run/docs/configuring/memory-limits#setting "
```

Comment:
The `CloudRunJob` block had cpu and memory as `ints` as well as a memory units field. 

I opted to convert this to the actual value strings that should be used, but I can swap this back if anyone feels strongly

Reply:
I'm fine with either, but I lean more towards your implementation to have one field each for memory and CPU vs. two fields for each.